src/observations/radec_mpc.jl

```diff
@@ -0,0 +1,497 @@
+abstract type AbstractObservation end
```

Comment:
I'm taking a closer look here, and I'm wondering whether this new observations interface allows querying a la Query.jl or DataFrames.jl as is shown in the Apophis notebooks? When dealing with a large number of observations coming from different objects and/or observatories, having a querying tool comes in pretty handy. For example, querying allowed us to filter optical astrometry coming from a single observatory which displayed a clear bias in the residuals. Maybe it's just a matter of defining a method to convert a `Vector{RadecMPC}` into a `DataFrame`?

Reply:
I think this is a good point, but I suggest to open an issue and leave it for another PR.

As a side remark on this proposal, I would suggest to have either `Query.jl` or `DataFrames.jl` as weakdeps in Julia 1.9, and for previous versions of Julia, exploit `Requires.jl`.

Reply:
Makes sense and I agree with the approach, so just opened an issue related to this; see #9